npc/custom/warper.txt

```diff
@@ -69,7 +70,19 @@ function Go {
 	set lastwarp$, getarg(0);
 	set lastwarpx, getarg(1,0);
 	set lastwarpy, getarg(2,0);
-	warp getarg(0),getarg(1,0),getarg(2,0);
+	for(.@i=0;.@i<getarraysize(.random_coordinates$);.@i++){
```

Comment:
If the array’s size is static for every iteration, store the size in a variable instead.

Reply:
This whole array is useless. Simply override the towns you want to randomize in the OnInit with zeroes. This way you do not have to compare it on each warp.

Reply:
the idea of this array is for the MVP maps while some servers block them and other make them warp-able to random coordinates i did try to see if there is way to check if a MVP-map didn't find that , and it maybe letting the server owner can choose the map him self would be better idea for some people would like to make the warp normal for orc maps and maybe disable some mini-bosses map like Valkyrie and Ghostring

Reply:
Yeah that's okay. I'm just saying that you dont need to do this logic on each warp.
Just do this parsing at the end of the OnInit event and you are fine. You parse the array once per reload and it should work the same way just being faster.

Reply:
i think i did get what you mean , making the check load once on server reload and not every-time the player click on the npc right ?

Reply:
exactly. :)